pheTransitionDates function initial commit

kjones13 opened the pull request:
@sokole first draft of the new phenology function, ready for initial reveiw

sokole commented:
@kjones13 
 - [ ] The convention we are using for function names is verbNoun. Like stackPlantPresence. Maybe `calcPheTransitionDates`? 
 - [ ] The convention we've adopted for this package is for functions to take list objects from neonUtilities as inputs. For example, in Dave's function, the argument is divDataList, because it gets sent the list object for plant diversity that is returned by neonUtilities. Here, maybe we could make the argument `pheDataList` and the function is expect the phe list object returned by neonUtilities. Then, rather than sending it `pheDat$phe_statusintensity`, you could just do pheDataList = pheDat in your example. 
 - [ ] You need to put the package prefix in front of functions used in your package that are not from base R. For example `dplyr::arrange()` instead of just `arrange`. If you run devtools::checks(), it will tell you all the instances of functions where we need to do this.

kjones13 commented:
@sokole in the last two commits I addressed all points above, changed name, added namespace, added functionality for using either data list or single table. Ready for re-review?

kjones13 commented:
@sokole this version of the estimatePheDurationByTag is flagging duplicates where none exist. I could use help troubleshooting.

kjones13 commented:
@sokole both functions work now. Ready to merge?

sokole commented:
@kjones13 awesome, I'll take a look and merge if everything looks good.

kjones13 commented:
Just checking in @sokole, it would be helpful to share these functions for the meeting today. Can we merge?
